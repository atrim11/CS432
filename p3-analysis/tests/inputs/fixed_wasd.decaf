int world;
bool print;

def int main() {
  int now;
  bool now1;
  bool faelse;
  bool help;
  int test;
  now1 = true;
  test = 2;
  now = (5 + 3 - 2);
  print_str("hello\\n \"\\\\world");
  test = -0xffff123abcdef;
  test = 123456789;
return 5;
}

def void wow (int a, int b, int c, bool d) {return;}

def void bad(int a, int b) {}
