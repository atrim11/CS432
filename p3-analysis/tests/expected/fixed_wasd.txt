Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 main : () -> int
 wow : (int, int, int, bool) -> void
 bad : (int, int) -> void
 world : int
 print : bool

  FuncDecl name="main" return_type=int parameters={} [line 4]
  SYM TABLE:

    Block [line 4]
    SYM TABLE:
     now : int
     now1 : bool
     faelse : bool
     help : bool
     test : int

  FuncDecl name="wow" return_type=void parameters={a:int,b:int,c:int,d:bool} [line 19]
  SYM TABLE:
   a : int
   b : int
   c : int
   d : bool

    Block [line 19]
    SYM TABLE:

  FuncDecl name="bad" return_type=void parameters={a:int,b:int} [line 21]
  SYM TABLE:
   a : int
   b : int

    Block [line 21]
    SYM TABLE:

